dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -114,35 +117,30 @@ void DynamicData::copy(const DynamicData& other)
   align_state_ = other.align_state_;
   strm_ = other.strm_;
   type_ = other.type_;
-  descriptor_ = other.descriptor_;
   item_count_ = other.item_count_;
 }
 
-DDS::ReturnCode_t DynamicData::get_descriptor(MemberDescriptor& value, MemberId id) const
+DDS::ReturnCode_t DynamicDataImpl::get_descriptor(DDS::MemberDescriptor*& value, MemberId id)
 {
-  DynamicTypeMember_rch dtm;
+  const DDS::DynamicType_var base_type = DDS::DynamicType::_duplicate(get_base_type(type_));
```

Comment:
This kind of reference counting doesn't follow the normal patterns.  For functions not defined in IDL, there are two options:
- Follow the same parameter passing / return type rules as if it was an IDL operation. In this case the caller should increment the reference count, not the callee.
- Use _vars directly for parameters and return types and let the _var constructors/assignment/etc take care of reference counting